module/VuFind/src/VuFind/Db/Service/SearchService.php

```diff
@@ -38,6 +43,28 @@
  * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
  * @link     https://vufind.org/wiki/development:plugins:database_gateways Wiki
  */
-class SearchService extends AbstractDbService implements SearchServiceInterface
+class SearchService extends AbstractDbService implements SearchServiceInterface, DbTableAwareInterface
 {
+    use DbTableAwareTrait;
+
+    /**
+     * Set invalid user_id values in the table to null; return count of affected rows.
+     *
+     * @return int
+     */
+    public function cleanUpInvalidUserIds(): int
+    {
+        $searchTable = $this->getDbTable('search');
+        $allIds = $this->getDbTable('user')->getSql()->select()->columns(['id']);
```

Comment:
This can be quite huge when each item in the array takes ~32 bytes of memory, but I suppose it'd be very rare to have a high number of users and low memory limit.

Reply:
I'm pretty sure that this is just constructing a subquery but not executing it.

Reply:
Oops, right. Misread it to read all ID's into a PHP array.